csrc/ops/utils.cpp

```diff
@@ -391,10 +409,24 @@ std::vector<IterDomain*> newOutputDomain(const std::vector<Val*>& vals) {
 
 TensorView* newOutputTV(const std::vector<Val*>& vals, DataType dtype) {
   auto out_domain = newOutputDomain(vals);
-  return IrBuilder::create<TensorView>(
+  auto* new_out = IrBuilder::create<TensorView>(
       IrBuilder::create<TensorDomain>(
           out_domain, TensorDomain::getContiguityFilledWith(out_domain, true)),
       dtype);
+
+  DeviceMesh new_mesh;
```

Comment:
I wonder if we can get into an issue where we introduced a resharding expression (which would later get an explicit `set` operation by insertResharding preseg pass), but it's probably dependent on where newOutputDomain is called.

Reply:
Yes, with this PR, `set` will try to propagate the input sharding to the output. I expect this to be worked around by
1. Not use `set`, and instead create the TensorDomain/TensorView by hand and then the op. 
2. Use `set` but remember to overwrite the DIDs. 

Overall, I don't think this is a perfect solution, and I'm all ears for alternatives.

Reply:
Yeah, I think we won't solve this until sharding propagation is revamped into something more intelligent. I think this is a good temp solution until we come up with something else.